sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/DocumentQueryExecutionContextFactory.java

```diff
@@ -370,75 +419,96 @@ public static <T> Flux<? extends IDocumentQueryExecutionContext<T>> createSpecia
-                        "very fast and have long runtimes. Please ensure you are using one of the above two filters" +
-                        "with you vector search query.");
-            }
-            // Validate the size of TOP or LIMIT against MaxItemSizeForVectorSearch
-            int maxLimit = Math.max(queryInfo.hasTop() ? queryInfo.getTop() : 0,
-                queryInfo.hasLimit() ? queryInfo.getLimit() : 0);
-            int maxItemSizeForVectorSearch = Math.max(Configs.getMaxItemCountForVectorSearch(),
-                qryOptAccessor.getMaxItemCountForVectorSearch(cosmosQueryRequestOptions));
-            if (maxLimit > maxItemSizeForVectorSearch) {
-                throw new NonStreamingOrderByBadRequestException(HttpConstants.StatusCodes.BADREQUEST,
-                    "Executing a vector search query with TOP or LIMIT larger than the maxItemSizeForVectorSearch " +
-                        "is not allowed");
+        if (hybridSearchQueryInfo!=null) {
+            // Validate the TOP for hybrid search queries
+            if (!hybridSearchQueryInfo.hasTake()) {
+                throw new HybridSearchBadRequestException(HttpConstants.StatusCodes.BADREQUEST,
+                    "Executing a hybrid or full text query without Top or Limit can consume a large number of RUs " +
+                        "very fast and have long runtimes. Please ensure you are using the above filter " +
+                        "with your hybrid or full text search query.");
             }
-            // Set initialPageSize based on the smallest of TOP or LIMIT
-            if (queryInfo.hasTop() || queryInfo.hasLimit()) {
-                int pageSizeWithTopOrLimit = Math.min(queryInfo.hasTop() ? queryInfo.getTop() : Integer.MAX_VALUE,
-                                           queryInfo.hasLimit() && queryInfo.hasOffset() ?
-                                               queryInfo.getLimit() + queryInfo.getOffset() : Integer.MAX_VALUE);
-                initialPageSize = pageSizeWithTopOrLimit;
+
+            if (hybridSearchQueryInfo.getSkip() >= hybridSearchQueryInfo.getTake()) {
+                throw new HybridSearchBadRequestException(HttpConstants.StatusCodes.BADREQUEST,
+                    "Executing a hybrid or full-text query with an offset(Skip) greater than or equal to limit(Take). " +
+                        "Please ensure limit is greater than offset.");
             }
-        }
 
-        // We need to compute the optimal initial page size for order-by queries
-        if (queryInfo.hasOrderBy()) {
-            int top;
-            if (queryInfo.hasTop() && (top = queryInfo.getTop()) > 0) {
-                int pageSizeWithTop = Math.min(
-                        (int)Math.ceil(top / (double)targetRanges.size()) * PageSizeFactorForTop,
-                        top);
+            int pageSize = hybridSearchQueryInfo.hasSkip() ? hybridSearchQueryInfo.getTake() - hybridSearchQueryInfo.getSkip() : hybridSearchQueryInfo.getTake();
```

Comment:
for the page size, should it be hybridSearchQueryInfo.getTake() `+ `hybridSearchQueryInfo.getSkip()?

Reply:
Done